pkg/controller/certificates/trigger/policies/gatherer.go

```diff
@@ -16,6 +16,201 @@ limitations under the License.
 
 package policies
 
+// In order to decide whether a certificate should be reissued or not, we want
+// to be gathering the "state of the world" regarding that particular
+// certificate, which is the entire purpose of DataForCertificate. Along with
+// the certificate's secret, DataForCertificate also returns two separate
+// certificate requests: the "current" and the "next" certificate request.
+//
+// To understand the roles of the "current" and "next" certificate requests, let
+// us look at three different scenarios: A, B and C.
+//
+// DIAGRAM (A1): in the first scenario, a user just created a Kubernetes
```

Comment:
I think you probably want to say cert-manager instead of Kubernetes.

Sadly, that will not help users, in some ways it's better to say Kubernetes.

Maybe:
```suggestion
// DIAGRAM (A1): in the first scenario, a user just created a cert-manager Kubernetes
```

It's hard to reason through the right order for such a thing.

Reply:
Fixed in 2361f355a, does it make more sense?